File: lightbulb-pickers/src/main/java/com/github/RooneyAndShadows/lightbulb/pickers/dialog/base/DialogPickerTriggerLayout.java
package com.github.RooneyAndShadows.lightbulb.pickers.dialog.base;

import android.graphics.drawable.Drawable;

public interface DialogPickerTriggerLayout {

    void attachTo(BaseDialogPickerView pickerView);

    void setTriggerIcon(Drawable icon, Integer color);

    void setTriggerText(String newText);

    void setTriggerErrorText(String errorText);

    void setTriggerHintText(String hintText);

    void setTriggerErrorEnabled(boolean errorEnabled);

    String getTriggerText();
}

File: lightbulb-pickers/src/main/java/com/github/RooneyAndShadows/lightbulb/pickers/dialog/base/DialogPickerTriggerLayout.kt
package com.github.rooneyandshadows.lightbulb.pickers.dialog.base

import android.graphics.drawable.Drawable

interface DialogPickerTriggerLayout {
    fun attachTo(pickerView: BaseDialogPickerView<*>)
    fun setEnabled(enabled: Boolean)
    fun setTriggerIcon(icon: Drawable?, color: Int?)
    fun setTriggerErrorText(errorText: String?)
    fun setTriggerHintText(hintText: String?)
    fun setTriggerErrorEnabled(errorEnabled: Boolean)
    var triggerText: String
}